Library/ViewModels/SetYourPasswordViewModelTests.swift

```diff
@@ -0,0 +1,56 @@
+import XCTest
+
+@testable import Library
+import ReactiveExtensions_TestHelpers
+
+final class SetYourPasswordViewModelTests: TestCase {
+  private let saveButtonIsEnabled = TestObserver<Bool, Never>()
+  private let contextLabelText = TestObserver<String, Never>()
+  private let newPasswordLabel = TestObserver<String, Never>()
+  private let confirmPasswordLabel = TestObserver<String, Never>()
+
+  func test_init() {
+    _ = self.makeSUT(with: "test@email.com")
+
+    XCTAssertNil(self.saveButtonIsEnabled.lastValue)
+  }
+
+  func test_saveButtonIsEnabledWhenFormIsValid() {
+    let sut = self.makeSUT()
+
+    sut.inputs.newPasswordFieldDidChange("")
+    sut.inputs.confirmPasswordFieldDidChange("")
+
+    self.saveButtonIsEnabled.assertLastValue(false)
+
+    sut.inputs.newPasswordFieldDidChange("somepassword")
+
+    self.saveButtonIsEnabled.assertLastValue(false)
+
+    sut.inputs.confirmPasswordFieldDidChange("somepass")
+
+    self.saveButtonIsEnabled.assertLastValue(false)
+
+    sut.inputs.newPasswordFieldDidChange("asdfasdf")
+    sut.inputs.confirmPasswordFieldDidChange("asdfasdf")
+
+    self.saveButtonIsEnabled.assertLastValue(true)
+  }
+
+  // MARK: - Helpers
+
+  private func makeSUT(with email: String = "", file _: StaticString = #filePath,
```

Comment:
Can we make this `makeViewModel`? System under test is a little ambiguous (just for readability).

Also probably don't need the `email` or `file` parameters, unless you see a use for them?

Can also write tests for these outputs:
  var contextLabelText: Signal<String, Never> { get }
  var newPasswordLabel: Signal<String, Never> { get }
  var confirmPasswordLabel: Signal<String, Never> { get }

Reply:
The viewModel needs to be configured with an email string in order for the contextLabel in the VC to show properly. 

Yeah i'll remove the `file` params. They were for my memoryleak test, but that doesn't look to be relevant now that I understand Signal a bit more 👍